### PR TITLE
Send better client version description including revision

### PR DESCRIPTION
**Pull request by gyunderscorebe**
## Original issue

#56
#115

## Description

Instead of sending version as integer and build type as a bit-packed integer,
send a string containing revision (with fallback to version number) and platform info in readable way.

Why not send this info in a more compact way (as some integers)? This way, with a string, it's easy to extend what goes into the description without breaking compatibility. (This PR itself _does_ break compatibility and increases protocol version number.)

Revision is taken from AF_REVISION macro. It can be defined on the command-line. I've added
an easy way to define it in Makefile, need to do something similar for Windows and Mac.

## How do we know this works?

I've tested this with ``make install AFREVISION=`git describe --tags` ``, server logs something like  `[192.168.1.250] ZZ|gy_be logged in (default), AC: revision version_1000-39-g37ebde6 (Linux, gcc)` when I connect.

## Check List

* [ ] AF_REVISION can be easily defined for make, used in `.travis.yml`.
* [ ] AF_REVISION can be easily defined for Visial Studio, used in `appveyor.yml`.
* [ ] AF_REVISION can be easily defined for XCode, used in `circle.yml`.
* [ ] current_version can handle strings
* [ ] download menu shows current version and version to download

### CODE REVIEW

File: source/src/client.cpp [hunk 1]
```diff
@@ -581,10 +581,44 @@ void c2sinfo(playerent *d)                  // send update to the server
     lastupdate = totalmillis;
 }
 
+static string versiondescription;
+
+string &getversiondescription()
+{
+    char platform[] =
+
+#ifdef _WIN32
```
Comment (lucasgautheron) [inline source/src/client.cpp]: WIN32 was fine, idk if _WIN32 is ever defined

File: source/src/client.cpp [hunk 2]
```diff
@@ -581,10 +581,44 @@ void c2sinfo(playerent *d)                  // send update to the server
...
+    "Windows"
+#elif __APPLE__
+    "Mac"
+#elif __linux__
+    "Linux"
+#else
+    "Unknown"
+#endif
+
+#ifdef __GNUC__
```
Comment (lucasgautheron) [inline source/src/client.cpp]: maybe look for __CLANG__ ? (or something like this).
Also I think clang defines __GNUC__...
And by the way, maybe add some informaton for windows: which compiler ? for msvc, https://msdn.microsoft.com/en-us/library/b0084kay.aspx (e.g. _MSC_FULL_VER)

File: source/src/client.cpp [hunk 3]
```diff
@@ -581,10 +581,44 @@ void c2sinfo(playerent *d)                  // send update to the server
...
+#endif
+
+    return versiondescription;
+}
+
 int getbuildtype()
 {
     return (isbigendian() ? 0x80 : 0 )|(adler((unsigned char *)guns, sizeof(guns)) % 31 << 8)|
-        #ifdef WIN32
+        #ifdef _WIN32
```
Comment (lucasgautheron) [inline source/src/client.cpp]: WIN32

**Comment by gyunderscorebe** [comment]
Looking at Travis, uh, we may have to remove "testing" tags for `git describe --tags` to work in a useful way.

File: source/src/client.cpp [hunk 4]
```diff
@@ -581,10 +581,44 @@ void c2sinfo(playerent *d)                  // send update to the server
     lastupdate = totalmillis;
 }
 
+static string versiondescription;
```
Comment (lucasgautheron) [inline source/src/client.cpp]: could be move into getversiondescription

**Comment by ScalaWilliam** [comment]
We need a release when this gets merged btw, since it's a breaking change.

**Comment by ScalaWilliam** [comment]
@lucasgautheron satisfied with this pr?

**Comment by lucasgautheron** [comment]
@ScalaWilliam yes but have a look at the checklist :)

**Comment by ScalaWilliam** [comment]
Next time one pr per checkbox 👍  I'm getting there with #165.... just about...

**Comment by ScalaWilliam** [comment]
Ignore that doesn't even make sense, I'm tired ;)

**Pull request merged**